File: baekjoon_all/22000+/boj_22856.cpp
// Solve 2024-02-12

#include <bits/stdc++.h>
using namespace std;

#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL); // boj_15552.cpp
#define SETPRECISION(n) cout << fixed;cout.precision(n); // boj_1008.cpp
#define SIZE(v) (int)v.size()
#define ALL(v) v.begin(),v.end()
using ll = long long;

struct Node{
    int left, right;
};

int main() {
    FASTIO;

    int n;
    cin >> n;

    Node tree[100001];

    for (int i = 0; i < n; i++) {
        int cur, left, right;
        cin >> cur >> left >> right;

        tree[cur].left = left;
        tree[cur].right = right;
    }

    int ans = (n - 1) * 2;

    for (int cur_node = 1; cur_node != -1; cur_node = tree[cur_node].right) {
        ans--;
    }

    cout << ++ans << '\n';

    return 0;
}

File: baekjoon_all/22000+/boj_22856.java
// Solve 2024-02-12

// 백준에 제출할 때는 class 이름을 Main으로 설정해야 한다.

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class boj_22856 {

    static class Node {
        int left, right;

        Node(int left, int right) {
            this.left = left;
            this.right = right;
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int n = Integer.parseInt(br.readLine());
        Node[] tree = new Node[n + 1];

        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            int cur = Integer.parseInt(st.nextToken());
            int left = Integer.parseInt(st.nextToken());
            int right = Integer.parseInt(st.nextToken());

            tree[cur] = new Node(left, right);
        }

        int ans = (n - 1) * 2;

        for (int curNode = 1; curNode != -1; curNode = tree[curNode].right) {
            ans--;
        }

        System.out.println(++ans);
        br.close();
    }

}
